File: src/test/java/br/com/blz/testjava/model/ProductTest.java
package br.com.blz.testjava.model;

import static org.junit.Assert.assertTrue;

import java.util.ArrayList;
import java.util.List;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import br.com.blz.testjava.enums.WarehouseTypeEnum;

public class ProductTest {
	
	@DisplayName("Product test is marketable.")
	@Test
	public void productIsMarketableTest() {
		Product product = new Product();
		product.setSku(101060L);
		product.setName("Novo Produto de Testes");
		
		Inventory inventory = new Inventory();
		
		List<Warehouse> warehouses = new ArrayList<Warehouse>();
		Warehouse warehouse1 = new Warehouse("PR", 10, WarehouseTypeEnum.PHYSICAL_STORE);
		Warehouse warehouse2 = new Warehouse("SP", 15, WarehouseTypeEnum.ECOMMERCE);
		warehouses.add(warehouse1);
		warehouses.add(warehouse2);
		inventory.setWarehouses(warehouses);
		product.setInventory(inventory);
		
		assertTrue(product.getIsMarketable().equals(Boolean.TRUE));
	}
	
	@DisplayName("Product test is not marketable.")
	@Test
	public void productIsNotMarketableTest() {
		Product product = new Product();
		product.setSku(101060L);
		product.setName("Novo Produto de Testes Not Marktable");
		
		Inventory inventory = new Inventory();
		
		List<Warehouse> warehouses = new ArrayList<Warehouse>();
		Warehouse warehouse1 = new Warehouse("PR", 0, WarehouseTypeEnum.PHYSICAL_STORE);
		Warehouse warehouse2 = new Warehouse("SP", 0, WarehouseTypeEnum.ECOMMERCE);
		warehouses.add(warehouse1);
		warehouses.add(warehouse2);
		inventory.setWarehouses(warehouses);
		product.setInventory(inventory);
		
		assertTrue(product.getIsMarketable().equals(Boolean.FALSE));
	}
	
}

File: src/test/java/br/com/blz/testjava/model/ProductTest.kt
package br.com.blz.testjava.model

import br.com.blz.testjava.enum.ProductType
import org.assertj.core.util.Lists
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInstance

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class ProductTest {
  private lateinit var warehouse:  Warehouse
  private lateinit var warehouse2: Warehouse
  private lateinit var product: Product

  @BeforeEach
  fun init() {
    warehouse2 = Warehouse("Rio de Janeiro", 5, ProductType.ECOMMERCE)
    warehouse = Warehouse("São Paulo", 2, ProductType.ECOMMERCE)
    product = Product(12345L, "Perfume", Inventory(Lists.list(warehouse, warehouse2)))
  }

  @Test
  fun isMarketableProductTrueTest() {
    Assertions.assertTrue(product.isMarketable())
  }

  @Test
  fun isMarketableProductFalseTest() {
    product.inventory.warehouses.map { warehouse -> warehouse.quantity = 0 }
    Assertions.assertFalse(product.isMarketable())
  }
}
